provider/ovh/ovh.go

```diff
@@ -390,9 +409,16 @@ func newOvhChange(action int, endpoints []*endpoint.Endpoint, zones []string, re
 			if e.RecordTTL.IsConfigured() {
 				change.TTL = int64(e.RecordTTL)
 			}
-			for _, record := range records {
-				if record.Zone == change.Zone && record.SubDomain == change.SubDomain && record.FieldType == change.FieldType && record.Target == change.Target {
-					change.ID = record.ID
+
+			// The Zone might have multiple records with the same target. In order to avoid applying the action to the
+			// same OVH record, we remove a record from the list when a match is found.
+			for i := 0; i < len(newRecords); i++ {
```

Comment:
This is a third nested for-each loop. Matching and removing records worth to consider optimising to reduce complexity and performance.

Something like that, just an idea, not a working code
```go
recordMap := make(map[string]ovhRecord, len(records))
	for _, rec := range records {
		key := fmt.Sprintf("%s|%s|%s|%s", rec.Zone, rec.SubDomain, rec.FieldType, rec.Target)
		recordMap[key] = rec
	}

// Use (datastructure) like the map to find and remove the record
			key := fmt.Sprintf("%s|%s|%s|%s", change.Zone, change.SubDomain, change.FieldType, change.Target)
			if rec, found := recordMap[key]; found {
				change.ID = rec.ID
				delete(recordMap, key)
			}

Reply:
It's a nice approach but there is a problem. The bug with the OVH provider was that multiple identical entries were added at each reconcile.
This for loop fixes that (no nore duplicate entries) and also fixes what has been added due to this bug...

So, the array might contain lots of identical records (same zone, subDomain, fieldType and target) and using a map will indexed only the last element and so those entries would not be cleaned.

Using a map would eventually clean up everything (I suppose) but i would take a huge amount of time.
The test (see below) has also been modified to take that bug into account.

I can do what you suggested if you think it's really needed. Meanwhile, I let is as is. Please advise :)